src/gobject.cc

```diff
@@ -200,20 +200,32 @@ static void GObjectClassDestroyed(const Nan::WeakCallbackInfo<GType> &info) {
 
     DestroyVFuncs(gtype);
 
-    auto persistentTpl = (Persistent<FunctionTemplate> *)
+    auto persistentTpl = (Nan::Persistent<FunctionTemplate> *)
         g_type_get_qdata (gtype, GNodeJS::template_quark());
-    auto persistentFn  = (Persistent<Function> *)
+    auto persistentFn  = (Nan::Persistent<Function> *)
         g_type_get_qdata (gtype, GNodeJS::function_quark());
-    delete persistentTpl;
+    persistentTpl->Reset();
     delete persistentFn;
 
     g_type_set_qdata (gtype, GNodeJS::template_quark(), NULL);
     g_type_set_qdata (gtype, GNodeJS::function_quark(), NULL);
     g_free(gtypePtr);
 }
 
-static void GObjectFallbackPropertyGetter(Local<v8::Name> property,
-                                            const v8::PropertyCallbackInfo<Value>& info) {
+#if defined(V8_MAJOR_VERSION) && (V8_MAJOR_VERSION > 12 || \
+    (V8_MAJOR_VERSION == 12 && defined(V8_MINOR_VERSION) && V8_MINOR_VERSION > 4))
+#define PROPERTY_CALLBACK_RETURN_TYPE v8::Intercepted
+#define PROPERTY_CALLBACK_RETURN(intercepted) return v8::Intercepted::intercepted
+#define PROPERTY_CALLBACK_INFO_TYPE v8::PropertyCallbackInfo<void>
+#else
+#define PROPERTY_CALLBACK_RETURN_TYPE void
+#define PROPERTY_CALLBACK_RETURN(intercepted) return
+#define PROPERTY_CALLBACK_INFO_TYPE v8::PropertyCallbackInfo<Value>
+#endif
```

Comment:
I replaced this with the [equivalent Nan abstractions](https://github.com/nodejs/nan/blob/main/doc/methods.md#api_nan_property_callback_info). This sort of thing is always handled by Nan, you don't need to rewrite them.